### ISSUE TITLE
Abstraction is not working

### ISSUE THREAD

**Issue by javrasya** [klarna/HiveRunner#48]
I try to build an `AbstractHiveTest`. But I can't use `@HiveSQL` annotation in abstract class. I get error such as;

``` console
java.lang.IllegalArgumentException: Failed to set field 'hiveShell' on 'com.bird.dwh.hadoop.hive.insertfact.InsertFactLocation2Test@213f8326': hiveShell
```

Here is some part of my abstract class;

``` java
@RunWith(StandaloneHiveRunner.class)
public abstract class AbstractHiveTest {

    @HiveSQL(files = {}, autoStart = false)
    HiveShellContainer hiveShell;

    ...
```

I follow the trace and I got somewhere that retrieves declared fields of child test classes. It only looks for child class fields to set `hiveShell`. I guess there should be a part missing to look into super class fields.

https://github.com/klarna/HiveRunner/blob/master/src/main/java/com/klarna/reflection/ReflectionUtils.java#L47

In here, there should also be something like I guess;

``` java
Field field = clazz.getSuperclass().getDeclaredFields()
```

**Comment by PelleUllberg** [comment]
Sorry for the late reply, I've been on vacation.

Yes, you're correct in that it does not find the annotated field in the super class. I've had this problem my self but solved it with an ugly workaround where the subclass has to declare the annotated HiveShell and then expose it via a getter to the super class.

I had a quick look at the reflection we're doing and it's a small mess :D I'll try to look at it but I'm in a tight spot at the moment. Feel free to file a pull request if you find the time to fix it your selves. I'll get around to it eventually since I also have some abstract test bases that would be improved if this would be fixed.

regards
/Pelle

**Comment by jorgenfalk** [comment]
I made a small recursive getField() method that picks the first matching field. There's a PR, but the build (sign-artifacts) on Travis failed due to some gpg issues:

gpg: no default secret key: unusable secret key

When I get some more time I'll try to see if I can fix the build.

**Comment by jorgenfalk** [comment]
I got a minor comment about the name of the abstract base test class that it should not end with 'Test' since it's not a test class. I renamed it to 'AnnotatedBaseTestClass'.
Also, since other builds has been working fine, I just try to rebuild this and see if the cert issues I had before with Travis is gone...

**Comment by PelleUllberg** [comment]
Fixed and released in HiveRunner 3.1.0. 
Closing.
